android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ApplicationExtension.kt

```diff
@@ -59,8 +59,10 @@ fun Application.lastSyncDateTime(): String {
   return lastSyncDate?.asString() ?: ""
 }
 
-fun <T> Application.loadConfig(id: String, clazz: Class<T>): T {
-  val json = assets.open(id).bufferedReader().use { it.readText() }
+fun <T> Application.loadConfig(id: String, clazz: Class<T>, data: Map<String, String?>): T {
+  var json = assets.open(id).bufferedReader().use { it.readText() }
+
+  data.entries.forEach { it.value?.let { v -> json = json.replace(it.key, v) } }
```

Comment:
Migrate this to `FormConfigUtil` class. You can overload the method or provided default value for argument `data`

Reply:
Renamed method. It actually does not represent form but a resource template i.e. IG/ImplementationGuide which would later be loaded from server